muonShieldOptimization/run_MufluxfixedTarget.py

```diff
@@ -20,6 +20,7 @@
 npot  = 5E13
 nStart = 0
 
+CharmdetSetup = 0 # 1 charm cross section setup, 0 muonflux setup
```

Comment:
This is unnecessary. You should be able to use `args.CharmdetSetup` everywhere. You can also give `ap.add_argument` an optional `default=0` argument.

Reply:
Ok, I can edit that. I saw that the other arguments are always stored in an ancillary variable, so I just assumed that this was the standard policy.

Reply:
I think it's just inherited from the way `getopt` did it. I would recommend using the `default=` option to `add_argument` and using `args.SomeArgument` directly if the scope allows.

Reply:
I see, thank you for the explanation. I have changed this part as you suggested

Reply:
Thanks!